src/PostgREST/Middleware.hs

```diff
@@ -51,16 +49,14 @@ import PostgREST.Request.Preferences
 import Protolude      hiding (head, toS)
 import Protolude.Conv (toS)
 
--- | Runs local(transaction scoped) GUCs for every request, plus the pre-request function
-runPgLocals :: AppConfig   -> M.HashMap Text JSON.Value ->
-               (ApiRequest -> ExceptT Error H.Transaction Wai.Response) ->
-               ApiRequest  -> ExceptT Error H.Transaction Wai.Response
-runPgLocals conf claims app req = do
+-- | Set local(transaction scoped) GUCs for every request, plus the pre-request function
+setPgLocals :: AppConfig -> M.HashMap Text JSON.Value -> ApiRequest
+  -> ExceptT Error H.Transaction ()
+setPgLocals conf claims req = do
```

Comment:
This is not really middleware, and does not really wrap the application. I renamed it to `setPgLocals` to reflect its function better, will probably move to `App` later on.